From: Dmitry Timoshkov <dmitry@baikal.ru>
Subject: [PATCH] user32: Don't check active window state when handling SC_MINIMIZE/SC_MAXIMIZE/SC_RESTORE.
Message-Id: <20211013132643.907686b179c62a4f26edf96b@baikal.ru>
Date: Wed, 13 Oct 2021 13:26:43 +0300

An application I'm working on routes messages to a worker thread, and checking
for an active window while processing WM_SYSCOMMAND in that thread breaks things.

Signed-off-by: Dmitry Timoshkov <dmitry@baikal.ru>
---
 dlls/user32/nonclient.c | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/dlls/user32/nonclient.c b/dlls/user32/nonclient.c
index 02e6ad92320..b47616bb639 100644
--- a/dlls/user32/nonclient.c
+++ b/dlls/user32/nonclient.c
@@ -1560,19 +1560,18 @@ LRESULT NC_HandleSysCommand( HWND hwnd, WPARAM wParam, LPARAM lParam )
         break;
 
     case SC_MINIMIZE:
-        if (hwnd == GetActiveWindow())
-            ShowOwnedPopups(hwnd,FALSE);
+        ShowOwnedPopups(hwnd,FALSE);
         ShowWindow( hwnd, SW_MINIMIZE );
         break;
 
     case SC_MAXIMIZE:
-        if (IsIconic(hwnd) && hwnd == GetActiveWindow())
+        if (IsIconic(hwnd))
             ShowOwnedPopups(hwnd,TRUE);
         ShowWindow( hwnd, SW_MAXIMIZE );
         break;
 
     case SC_RESTORE:
-        if (IsIconic(hwnd) && hwnd == GetActiveWindow())
+        if (IsIconic(hwnd))
             ShowOwnedPopups(hwnd,TRUE);
         ShowWindow( hwnd, SW_RESTORE );
         break;

-- 
2.31.1

